### PR TITLE
Bah 8940 transform

### PR DESCRIPTION
**Pull request by zurbergram**
## Description
Transform data from form data unemployability to vets-json-schema form 8940

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/disability-benefits/all-claims/submit-transformer.js [hunk 1]
```diff
@@ -354,6 +366,36 @@ export function transform(formConfig, form) {
     }
     return clonedData;
   };
+
+  const addForm8940 = formData => {
+    const clonedData = _.cloneDeep(formData);
+    const unemployability = clonedData.unemployability;
```
Comment (aaronjlech) [inline src/applications/disability-benefits/all-claims/submit-transformer.js]: you could do a pluck here `const { unemployability } = clonedData`. Little shorter 🤷‍♂️

File: src/applications/disability-benefits/all-claims/submit-transformer.js [hunk 2]
```diff
@@ -354,6 +366,36 @@ export function transform(formConfig, form) {
...
+
+    if (unemployability) {
+      const disabilities = getDisabilities(formData);
+
+      clonedData.form8940 = {
+        unemployability: {
+          ...unemployability,
+          disabilityPreventingEmployment: disabilities
+            .filter(disability => disability.unemployabilityDisability)
+            .flatMap(disability => getDisabilityName(disability))
```
Comment (aaronjlech) [inline src/applications/disability-benefits/all-claims/submit-transformer.js]: this could just be `.map()`

**Review by aaronjlech — APPROVED** [review]
Just a few suggestions! lgtm otherwise

**Comment by zurbergram** [comment]
@department-of-veterans-affairs/code-reviewers-for-external-teams-group - Could I get a code review on this at your earliest convenience? Thanks!

**Review by jbalboni — APPROVED** [review]

File: src/applications/disability-benefits/all-claims/submit-transformer.js [hunk 3]
```diff
@@ -67,11 +67,14 @@ export function transformProviderFacilities(providerFacilities) {
...
+/**
+ * Returns an array of disabilities pulled from ratedDisabilties, newDisabilities, newPrimaryDisabilities and newSecondaryDisabilities
+ * @param {object} formData
+ */
+function getDisabilities(formData) {
   // Assumes we have only selected conditions at this point
-  // TODO: Filter by disabilityIsSelected
   const claimedConditions = formData.ratedDisabilities
-    ? formData.ratedDisabilities.map(d => d.name.toLowerCase())
+    ? formData.ratedDisabilities
```
[the remark this answers is not in the document]
Comment (zurbergram) [reply src/applications/disability-benefits/all-claims/submit-transformer.js]: yes, didn't know that was a thing in javascript

**Pull request merged**